src/NLog/Internal/TargetWithFilterChain.cs

```diff
@@ -92,17 +94,21 @@ internal void PrecalculateStackTraceUsage()
 
             // find all objects which may need stack trace
             // and determine maximum
-            foreach (var item in ObjectGraphScanner.FindReachableObjects<IUsesStackTrace>(this))
+            // only the target can have IUsesStackTrace
+            if (Target != null)
```

Comment:
AFAIK that isn't needed. This call is invoked for every targerwithchain.  But please correct me if I missed something.